platform-modules/content-manager/src/main/java/org/ekstep/content/operation/finalizer/PublishFinalizer.java

```diff
@@ -1416,4 +1427,25 @@ protected String getItemsetPreviewUrl(Node node) throws Exception {
     		return null;
     }
 
+	 public void setContentAndCategoryTypes(Map<String, Object> input)  {
+		String contentType = (String)input.get("contentType");
+		 System.out.println(contentType);
+		 System.out.println(contentPrimaryCategoryString);
+		String primaryCategory = (String) input.get("primaryCategory");
+		String updatedContentType = "";
+		String updatedPrimaryCategory = "";
+		if(StringUtils.isNotBlank(contentType) && StringUtils.isBlank(primaryCategory)) {
+			updatedContentType = contentType;
+			updatedPrimaryCategory = contentPrimaryCategoryString.get(contentType);
+		} else if(StringUtils.isBlank(contentType) && StringUtils.isNotBlank(primaryCategory)) {
+			updatedContentType = contentPrimaryCategoryString.entrySet().stream().filter(entry -> StringUtils.equalsIgnoreCase(entry.getValue(), primaryCategory))
+					.map(entry -> entry.getKey()).findFirst().orElse("");
+			updatedPrimaryCategory = contentPrimaryCategoryString.get(contentType);
```

Comment:
Set as updatedPrimaryCategory = primaryCategory
If primaryCategory is not blank what is the need to update it.

Reply:
WIll do